flink-tests/src/test/java/org/apache/flink/test/checkpointing/SavepointITCase.java

```diff
@@ -431,36 +430,48 @@ public void invoke(Long value) {
                                 .build());
         cluster.before();
         try {
-            final JobID jobID1 = new JobID();
-            jobGraph.setJobID(jobID1);
-            cluster.getClusterClient().submitJob(jobGraph).get();
-            CommonTestUtils.waitForAllTaskRunning(cluster.getMiniCluster(), jobID1, false);
+            final JobID firstJobId = new JobID();
+            final JobGraph firstJobGraph = InstantiationUtil.clone(jobGraph);
+            firstJobGraph.setJobID(firstJobId);
+            cluster.getClusterClient().submitJob(firstJobGraph).get();
+            CommonTestUtils.waitForAllTaskRunning(cluster.getMiniCluster(), firstJobId, false);
             // wait for some records to be processed before taking the checkpoint
             counter.get().await();
-            final String firstCheckpoint = cluster.getMiniCluster().triggerCheckpoint(jobID1).get();
-
-            cluster.getClusterClient().cancel(jobID1).get();
-            jobGraph.setSavepointRestoreSettings(
+            final String firstCheckpoint =
+                    cluster.getMiniCluster().triggerCheckpoint(firstJobId).get();
+            cluster.getClusterClient().cancel(firstJobId).get();
+            CommonTestUtils.waitForJobStatus(
```

Comment:
I don't fully understand why we need it here. if we clone JobGraph then everything works perfectly fine even without this waiting. It seems for me that the awaiting on `cancel's future` is enough to be sure that the next submit with the new JobGraph will be successful. Did I miss some rare race condition?

Reply:
This would fix the test as well, even without cloning the `JobGraph`, because after this point the mutations wouldn't affect the dispatcher.

Reply:
Since you cloned the object inside of submit. Do I understand correctly, that we can avoid `waitForJobStatus`? (I understand that it also makes sense, but I just don't want to overcomplicate the code if it is not necessary)